pkg/kubelet/apis/config/types.go

```diff
@@ -220,6 +220,8 @@ type KubeletConfiguration struct {
 	// cpuCFSQuota enables CPU CFS quota enforcement for containers that
 	// specify CPU limits
 	CPUCFSQuota bool
+	// CPUCFSQuotaPeriod sets the CPU CFS quota period value, cpu.cfs_period_us, defaults to 100ms
```

Comment:
I'm super confused about this one and can't figure this one out by myself: `cpu.cfs_period_us` default limit called "100ms" but [actually](https://www.kernel.org/doc/html/latest/scheduler/sched-bwc.html#management) is microseconds:
> `cpu.cfs_period_us`: the length of a period (in microseconds)

In the change I'm commenting author clearly acted under assumptions that milliseconds are used in k8s and now it's all over the test cases and comments. But I wonder where that 1000x disperancy between k8s value and Linux value comes from? And does it exist, or k8s default is actually a microsecond, same as linux? Unless we multiple [that](https://github.com/kubernetes/kubernetes/blob/37761a3/pkg/kubelet/cm/helpers_linux.go#L47-L48) default value of `QuotaPeriod` by 1000 somewhere, converted to `time.Duration` it's 100 microseconds and not milliseconds.

cc @vishh @sjenning and especially @hjacobs: I hope you all have some context about it and can tell me why I'm wrong and k8s code comments are right.

Reply:
yes cfs_period_us is in µs, that's why it has `_us` suffix
Maybe https://go.dev/play/p/65fG1OmLFYN helps to understand.
As far as I remember, I think the time.Duration is only used if you want to set it via kubelet flag/config.

Reply:
The comment in that MR says "ms"." In contrast, it better be "microseconds", I assume. I can send MR to change it as well as explicit millisecond mentions in the current code, whereas it should be microseconds. I'll ping you in it.

In this MR, you wrote milliseconds duration in multiple places in defaults and tests `CPUCFSQuotaPeriod: metav1.Duration{Duration: 100 * time.Millisecond},`, I wonder if that should be changed to microseconds to match the actual default value of 100 microseconds?

Also, the discussion which led to this MR has multiple people talking about changing the value from 100 milliseconds to 5/25 milliseconds for testing. Still, it doesn't make sense unless everyone were altering the source code in microseconds, thinking it's milliseconds. @hjacobs even did [a talk](https://youtu.be/eBChCFD9hfs?t=870) about it talking about the default value as if it was in milliseconds. Your code introducing milliseconds and not microseconds (so 1000x) contributed to my confusion, and I wonder if it's intentional or if you are mistaken, and it should be changed microseconds to change the real default.

The only way current code would make sense to me is if the value you feed to k8s (100 milliseconds) is divided by 1000 at some point in time and then matches the real Linux default of 100 microseconds, but I haven't found such a place yet, and I would doubt that's the case.

Reply:
@szuecs, the simplest form of the question I can form is the following.

Is the 100ms value of `CPUCFSQuota` result in the same k8s behaviour as the default Linux value of 100 microseconds, or is it 1000 times higher as its milliseconds versus microseconds in Linux?

Reply:
I was wrong, and the original code was correct. 100ms is indeed a default value in Linux, and I got confused by conversions in two places in code that had no comments about them. PR #112123 adds comments to the code added here.